consensus/src/tusk.rs

```diff
@@ -263,6 +262,9 @@ impl<PublicKey: VerifyingKey> Consensus<PublicKey> {
                     certificate: x,
                     consensus_index,
                 });
+
+                // Increase the global consensus index.
+                consensus_index += 1;
```

Comment:
Ok, so if I understand correctly this was starting the index for the first certificate from 1 and not from 0, right? What about when the `store.read_last_consensus_index()` is called?

https://github.com/MystenLabs/narwhal/blob/976355d2ff75aefc6b7ec7fd32742d3d7188c4e0/consensus/src/tusk.rs#L122

I assume we want to continue then from the "next" value of the consensus index. So for example, if we last stored the index `5` , then we want to continue from `6`. Would that be guaranteed as well if you change the index to that point here?

Reply:
Also, the fact that no test broke by making this change tells me that probably we don't have coverage, could you please add at least the simplest case that could verify/catch this?

Reply:
No tests broke because it is not a bug per se. The problem is that the client (not in the Narwhal codebase) expected sequence numbers starting at '1' while narwhal started its sequence numbers at '0'. A different client (other than Sui) may very well start its sequence numbers as '0', or 'x'.

Reply:
More intriguing however is why this bug wasn't triggered in a narwhal test (it was triggered in a test in Sui): #191